src/components/Forms/ReviewForm.tsx

```diff
@@ -0,0 +1,175 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { useEffect, useMemo, useState } from 'react';
+import { IoStarOutline } from 'react-icons/io5';
+import { IoStar } from 'react-icons/io5';
+import { useAppDispatch, useAppSelector } from '../../redux/hooks/hooks';
+import {
+	reviewFormSchema,
+	reviewFormSchemaType,
+} from '../../validations/products/reviewFromValidations';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { createReview } from '../../redux/features/CreateReviewSlice';
+import { DynamicData } from '../../@types/DynamicData';
+import { fetchReview } from '../../redux/features/getReviewSice';
+import fetchInfo from '../../utils/userDetails';
+import { editReview } from '../../redux/features/editReviewSlice';
+interface reviewFormType {
+	id: string;
+	accessToken: string;
+	handleReviewform: () => void;
+	successMessage: (message: string) => void;
+	Erromesage: (string: string) => void;
+}
+function ReviewForm(props: reviewFormType) {
+	const [isEditMode, setIsEditMode] = useState(false);
+	const [rated, setRated] = useState(0);
+	const [reviewId, setReviewId] = useState('');
+	const dispatch = useAppDispatch();
+	const {
+		register,
+		handleSubmit,
+		setValue,
+		formState: { errors },
+	} = useForm<reviewFormSchemaType>({
+		resolver: zodResolver(reviewFormSchema),
+	});
+	const data = useAppSelector(
+		(state) => state.fetchReview.data[state.fetchReview.data.length - 1]?.data,
+	);
+	const onSubmit: SubmitHandler<reviewFormSchemaType> = async (
+		data: reviewFormSchemaType,
+	) => {
+		try {
+			const { ratings, feedBack } = data;
+			const token = props.accessToken;
+			const productId = props.id;
+			if (isEditMode) {
+				const res = await dispatch(
+					editReview({ reviewId, feedBack, ratings, token }),
```

Comment:
for this editReview slice it shouldn't be receiving this token props since we're configure them in headers for every request in axios interceptor

Reply:
@leandreAlly  it is removed